File: Daily Problems/Medium/1233. Remove Sub-Folders from the Filesystem/solution.py
# Python solution for 1233. Remove Sub-Folders from the Filesystem

def removeSubfolders(self, folder: List[str]) -> List[str]:
    folder.sort()
    result = [folder[0]]
    
    for current in folder[1:]:
        last_parent = result[-1]
        
        # Check if `current` is a subfolder of `last_parent`
        if not (current.startswith(last_parent) and current[len(last_parent)] == '/'):
            result.append(current)
    
    return result

File: Daily Problems/Medium/1233. Remove Sub-Folders from the Filesystem/solution.R
# R solution for 1233. Remove Sub-Folders from the Filesystem

removeSubfolders <- function(folder) {
  folder <- sort(folder)
  result <- list(folder[1])
  
  for (i in 2:length(folder)) {
    current <- folder[i]
    last_parent <- tail(result, 1)[[1]]
    
    # Check if `current` is a subfolder of `last_parent`
    if (!(startsWith(current, last_parent) && substr(current, nchar(last_parent) + 1, nchar(last_parent) + 1) == "/")) {
      result <- append(result, current)
    }
  }
  
  return(unlist(result))
}